src/DataLoader/Extensions/DisposeWithNextLoad/SequenceDisposableGroup.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Chinook.DataLoader.DisposeWithNextLoad
+{
+	/// <summary>
+	/// This class is a container of <see cref="IDisposable"/> objects that automatically disposes its items when necessary.
+	/// It associates each disposable object with the <see cref="IDataLoaderRequest"/> that created them and keeps track of that request.
+	/// When items from a newer request are added, the items associated to a previous <see cref="IDataLoaderRequest"/> are disposed.
```

Comment:
Perhaps we should add to the description that this class is not thread-safe

Reply:
done :)